src/plone/restapi/serializer/collection.py

```diff
@@ -14,16 +16,19 @@ class SerializeCollectionToJson(SerializeToJson):
 
     def __call__(self):
         result = super(SerializeCollectionToJson, self).__call__()
-        portal = getSite()
-        result['member'] = [
-            {
-                '@id': '{0}/{1}'.format(
-                    portal.absolute_url(),
-                    '/'.join(member.getPhysicalPath())
-                ),
-                'title': member.title,
-                'description': member.description
-            }
-            for member in self.context.results()
-        ]
+
+        # XXX: collection.results() claims to return an IContentListing
```

Comment:
Ah, you're right - the `@@contentlisting` view does indeed call `results()`, but with different defaults for its keyword arguments - specifically `batch=False`. That's why I was seeing different results.

So to summarize:
- `collection.results(batch=False)` returns an `IContentListing`
- `collection.results()` wraps that content listing in a `Batch` before returning it

I'll update the comment, and simplify the code to something along the lines of `listing = self.context.results(batch=False)`